### PR TITLE
helm: Refactoring labels and support additional labels

### PR DESCRIPTION
**Pull request by nlamirault**
**What this PR does / why we need it**:

* Refactoring Kubernetes labels (Like https://helm.sh/docs/chart_best_practices/labels/ and https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/)
* Use selector labels (like a Helm chart create using Helm CLI)
* Support additional labels to add to all metadata

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by aramase — COMMENTED** [review]
@nlamirault Thank you for the PR.

Please update the yamls in the [manifest_staging/](https://github.com/kubernetes-sigs/secrets-store-csi-driver/tree/main/manifest_staging/) folder, where we host the staging charts and deployment yamls. All the yaml changes will then be promoted into the released charts folder (`charts/`) with the next release.

**Comment by aramase** [comment]
/retitle helm: Refactoring labels and support additional labels

**Comment by nlamirault** [comment]
@aramase done.

File: manifest_staging/charts/secrets-store-csi-driver/templates/_helpers.tpl [hunk 1]
```diff
@@ -27,16 +27,27 @@ If release name contains chart name it will be used as a full name.
...
 {{- end -}}
 
+{{/*
+Selector labels
+*/}}
+{{- define "sscd.selectorLabels" }}
+app.kubernetes.io/name: {{ include "sscd.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+app: {{ template "sscd.name" . }}
+{{- end }}
```
Comment (nilekhc) [inline manifest_staging/charts/secrets-store-csi-driver/templates/_helpers.tpl:49]: Are these selectorLabels specific to any scenario? If so, perhaps we could get it from Values file and make it generic?

**Comment by aramase** [comment]
@nlamirault Could you take a look at Nilekh's comment and address the failing tests? If this PR is not being actively worked on I would recommend we `/close` this and reopen when ready to review again.

File: manifest_staging/charts/secrets-store-csi-driver/templates/_helpers.tpl (hunk 1 above)
Comment (nlamirault) [reply manifest_staging/charts/secrets-store-csi-driver/templates/_helpers.tpl:49]: Not really specific labels. They are standard labels (except `app`), created for example by the Helm CLI with the `helm create newchart` command.
Ex: https://github.com/kubernetes/ingress-nginx/blob/main/charts/ingress-nginx/templates/_helpers.tpl#L98

**Comment by nlamirault** [comment]
Tests seems failed on cloud provider :(
On kind i test like that : 

```
$ make e2e-helm-deploy-release
NOTES:
The Secrets Store CSI Driver is getting deployed to your cluster.

To verify that Secrets Store CSI Driver has started, run:

  kubectl --namespace=kube-system get pods -l "app=secrets-store-csi-driver"

Now you can follow these steps https://secrets-store-csi-driver.sigs.k8s.io/getting-started/usage.html
to create a SecretProviderClass resource, and a deployment using the SecretProviderClass.
```

**Comment by nilekhc** [comment]
> Tests seems failed on cloud provider :( On kind i test like that :
> 
> ```
> $ make e2e-helm-deploy-release
> NOTES:
> The Secrets Store CSI Driver is getting deployed to your cluster.
> 
> To verify that Secrets Store CSI Driver has started, run:
> 
>   kubectl --namespace=kube-system get pods -l "app=secrets-store-csi-driver"
> 
> Now you can follow these steps https://secrets-store-csi-driver.sigs.k8s.io/getting-started/usage.html
> to create a SecretProviderClass resource, and a deployment using the SecretProviderClass.
> ```

Tests with `e2e-provider` aren't cloud provider specific. They seem to be failing due to incorrect labels. Could you please verify helm chart?
```
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(DaemonSet.spec.template): unknown field "labels" in io.k8s.api.core.v1.PodTemplateSpec
helm.go:84: [debug] error validating "": error validating data: ValidationError(DaemonSet.spec.template): unknown field "labels" in io.k8s.api.core.v1.PodTemplateSpec
```

**Comment by nlamirault** [comment]
I try with kubeval : 

```
❯  helm template charts/secrets-store-csi-driver --namespace kube-system --wait --timeout=5m -v=5 --debug \
        --set linux.image.pullPolicy="IfNotPresent" \
        --set windows.image.pullPolicy="IfNotPresent" \
        --set linux.image.repository=gcr.io/k8s-staging-csi-secrets-store/driver \
        --set linux.image.tag=v1.0.0-e2e-02056255 \
        --set windows.image.repository=gcr.io/k8s-staging-csi-secrets-store/driver \
        --set windows.image.tag=v1.0.0-e2e-02056255 \
        --set linux.crds.image.repository=gcr.io/k8s-staging-csi-secrets-store/driver-crds \
        --set linux.crds.image.tag=v1.0.0-e2e-02056255 \
        --set linux.crds.annotations."myAnnotation"=test \
        --set windows.enabled=true \
        --set linux.enabled=true \
        --set syncSecret.enabled=true | kubeval
install.go:173: [debug] Original chart version: ""
install.go:190: [debug] CHART PATH: /home/nicolas/Projects/Forks/secrets-store-csi-driver/charts/secrets-store-csi-driver

PASS - secrets-store-csi-driver/templates/serviceaccount.yaml contains a valid ServiceAccount
PASS - secrets-store-csi-driver/templates/role-syncsecret.yaml contains a valid ClusterRole
PASS - secrets-store-csi-driver/templates/role.yaml contains a valid ClusterRole
PASS - secrets-store-csi-driver/templates/role-syncsecret_binding.yaml contains a valid ClusterRoleBinding
PASS - secrets-store-csi-driver/templates/role_binding.yaml contains a valid ClusterRoleBinding
PASS - secrets-store-csi-driver/templates/secrets-store-csi-driver-windows.yaml contains a valid DaemonSet
PASS - secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml contains a valid DaemonSet
PASS - secrets-store-csi-driver/templates/csidriver.yaml contains a valid CSIDriver
PASS - secrets-store-csi-driver/templates/crds-upgrade-hook.yaml contains an empty YAML document
PASS - secrets-store-csi-driver/templates/crds-upgrade-hook.yaml contains a valid ServiceAccount
PASS - secrets-store-csi-driver/templates/keep-crds-upgrade-hook.yaml contains an empty YAML document
PASS - secrets-store-csi-driver/templates/keep-crds-upgrade-hook.yaml contains a valid ServiceAccount
PASS - secrets-store-csi-driver/templates/crds-upgrade-hook.yaml contains a valid ClusterRole
PASS - secrets-store-csi-driver/templates/keep-crds-upgrade-hook.yaml contains a valid ClusterRole
PASS - secrets-store-csi-driver/templates/crds-upgrade-hook.yaml contains a valid ClusterRoleBinding
PASS - secrets-store-csi-driver/templates/keep-crds-upgrade-hook.yaml contains a valid ClusterRoleBinding
PASS - secrets-store-csi-driver/templates/crds-upgrade-hook.yaml contains a valid Job
PASS - secrets-store-csi-driver/templates/keep-crds-upgrade-hook.yaml contains a valid Job
```

**Comment by aramase** [comment]
Closing this PR due to inactivity. Please feel free to reopen if this is still being worked on.

/close

**Pull request closed without merging**